src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.ConcurrentAccessDetection.netcoreapp.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Common.System;
+using Xunit;
+
+namespace Generic.Dictionary
+{
+    public class DictionaryConcurrentAccessDetectionTests
+    {
+        [Fact]
+        public static void Add_DictionaryConcurrentAccessDetection_NullComparer_ValueTypeKey()
+        {
+            Thread customThread = new Thread(() =>
+            {
+                Dictionary<int, int> dic = new Dictionary<int, int>();
+                dic.Add(1, 1);
+
+                //break internal state
+                var entriesType = dic.GetType().GetField("_entries", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
Could you please extract common code (eg the reflection code)

Reply:
ok